### PR TITLE
nfc: Remove nfc platform from nrfxlib

### PR DESCRIPTION
**Pull request by KAGA164**
nfc_platform moved to NCS.

Signed-off-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

### CODE REVIEW

**Comment by KAGA164** [comment]
@kapi-no `select NRFX_NFCT` moved to NCS

File: nfc/Kconfig [hunk 1]
```diff
@@ -4,58 +4,12 @@
...
-config NRFXLIB_NFC
-	bool
-	prompt "Enable NFC"
-	select NRFX_NFCT
-	help
-	  Enable NFC functionality
-
-if NRFXLIB_NFC
-
 comment "Common NFC module configuration"
```
Comment (anangl) [inline nfc/Kconfig]: This does not seem to be needed anymore.

Comment (KAGA164) [reply nfc/Kconfig]: Ok, removed.

File: nfc/Kconfig [hunk 2]
```diff
@@ -4,58 +4,10 @@
...
-	  Enable NFC functionality
-
-if NRFXLIB_NFC
-
-comment "Common NFC module configuration"
-
-config NFC_T2T_LIB_ENABLED
+config NFC_T2T_NRFXLIB
 	bool
 	prompt "Enable NFC Type 2 Tag library"
```
Comment (anangl) [inline nfc/Kconfig]: On the occasion, you could also get rid of these `prompt`s here.
```
bool "Enable NFC Type 2 Tag library"
```

Comment (KAGA164) [reply nfc/Kconfig]: Ok changed.

**Pull request merged**